_posts/2019-06-07-conventions-and-rubocop.markdown

```diff
@@ -0,0 +1,112 @@
+make it harder to port code from one platform to the next.  It was a
+natural extension from there to note that certain practices also made
+the code harder to read - and therefore maintain.
+
+Lint was enthusiastically embraced, and the underlying idea has since
+been repeated over and over again - at this point, it's fair to say
+that a language can't be considered to have arrived until it has,
+among other things, it's own Lint style tool.  This is important
+because these tools will often wind up as de-facto archives of the
+learned experiences of developers using the language, codifying what
+tends to make code generally readable to the workaday programmer, and
+what does the opposite.  Constructs that are obscure or unreadable,
+and constructs that have proven to be rich sources of bugs, will be
+accordingly documented and flagged - often with justifications for
+why.  Reading this documentation, if it's available, can be an
+excellent way to learn how to write solid, idiomatic code in a given
+language.  If you're learning a new language, such style guides are a
+natural next stop after you've mastered the basics.
+
+# Rubocop
+
+If you don't know by now,
+[Rubocop](https://github.com/rubocop-hq/rubocop) is the de-facto Lint
+equivalent for Ruby code.  The [Style
+Guide](https://github.com/rubocop-hq/ruby-style-guide) is an excellent
+example of communally created conventions, complete with
+justifications and links to ancillary documentation.   Like any self
+respecting Lint style tool, it recognizes that programmers love to
+disagree, and allows you to set up your own Ruleset or
+[disable](https://rubocop.readthedocs.io/en/latest/configuration/)
+rules you disagree with, either globally, or on a case by case basis.
+
+If you're using Git, then it's straightforward to add Rubocop checking
+to your pre-commit hooks, making style checks an automatic part of
+every developer's workflow.  This is probably better than integrating
+it into something more centralized, like a Jenkins server - it's best
+that code doesn't make it into your repository unless it conforms to
+conventions, rather than flagging it after the fact.
+
+# Getting over the hump
+
+If you're starting out with an existing code base, then as soon as you
+integrate Rubocop into the picture, you'll get dozens if not hundreds
+of style violations.  No worries, Rubocop has you covered.  The handy
+[autocorrect](https://rubocop.readthedocs.io/en/latest/auto_correct/)
+mode will fix violations automatically.  If that's a little aggressive
+for your tastes, then you can just generate a
+[rubocop_todo.yml](https://rubocop.readthedocs.io/en/latest/configuration/#automatically-generated-configuration)
+file that will disable checking for all *existing* violations while
+still flagging any new violations going forward.  You can then circle
+back and fix up your existing code, treating it as a part of your
+existing technical debt.
```

Comment:
@rdormer Could you add a section about `standardrb`? https://github.com/testdouble/standard